### PR TITLE
Log cache-hit message in terminal, rather than in actual log output file

### PR DESCRIPTION
**Pull request by mehulkar**

### CODE REVIEW

File: cli/internal/runcache/runcache.go [hunk 1]
```diff
@@ -193,6 +193,7 @@ func (tc TaskCache) RestoreOutputs(ctx context.Context, terminal *cli.PrefixedUi
 		terminal.Output(fmt.Sprintf("cache hit, suppressing output %s", ui.Dim(tc.hash)))
 	case util.FullTaskOutput:
 		logger.Debug("log file", "path", tc.LogFileName)
+		terminal.Output(fmt.Sprintf("cache hit, replaying output %s", ui.Dim(tc.hash)))
```
Comment (mehulkar) [inline cli/internal/runcache/runcache.go:196]: This is still `cli.PrefixedUi`, so the `package:task` prefix will still be there.

**Review by gsoltis — APPROVED** [review]
I think this is ok because we have a pending global-hash-key change. If this somehow ends up not in the same release as that change, we'll need another one to avoid duplicating cache-hit messages.

**Comment by mehulkar** [comment]
> I think this is ok because we have a pending global-hash-key change. If this somehow ends up not in the same release as that change, we'll need another one to avoid duplicating cache-hit messages.

For my understanding, will this actually cause cache missees? The inputs aren't changing from this PR, just the outputs, so it should any existing caches should remain valid? 

In any case, I discovered while working on #2126, that `OutputWriter` still needs the prefix for the stdout writer, so I will close this for now.

**Comment by gsoltis** [comment]
@mehulkar the issue isn't that it will cause misses, it's that it won't: logs created w/ previous versions will be restored and still have the cache hit line in the log, and this change prints the cache hit lines separately, then replays the log, resulting in two cache hit lines

**Pull request closed without merging**